integration-azure-oauth-test/integration-azure-oauth-test-docker/src/test/it/com/xwiki/entraid/test/ui/EntraIdIT.java

```diff
@@ -0,0 +1,167 @@
+    {
+        setup.createUser(FIRST_USER_NAME, PASSWORD, setup.getURLToNonExistentPage(), "first_name", "Jon", "last_name",
+            "Snow");
+
+        setup.createUser(SECOND_USER_NAME, PASSWORD, setup.getURLToNonExistentPage(), "first_name", "Elsa", "last_name",
+            "Ice");
+
+        // Create a group and add both users.
+        setup.loginAsSuperAdmin();
+        GroupsPage groupsPage = GroupsPage.gotoPage();
+        groupsPage.addNewGroup(GROUP_NAME);
+        EditGroupModal devsGroupModal = groupsPage.clickEditGroup(GROUP_NAME);
+        devsGroupModal.addUsers(FIRST_USER_NAME, SECOND_USER_NAME);
+        groupsPage = GroupsPage.gotoPage();
+        assertEquals("2", groupsPage.getMemberCount(GROUP_NAME));
+
+        // By default the minimal distribution used for the tests doesn't have any rights setup. Let's create an Admin
+        // user part of the Admin Group and make sure that this Admin Group has admin rights in the wiki. We could also
+        // have given that Admin user the admin right directly but the solution we chose is closer to the XS
+        // distribution.
+        setup.setGlobalRights("XWiki.XWikiAdminGroup", "", "admin", true);
+        setup.createAdminUser();
+        setup.loginAsAdmin();
+
+        // Modify the tenant ID to update the OIDC configuration so that the Microsoft login page is displayed.
+        setup.updateObject(ENTRAID_CONFIGURATION_REFERENCE, ENTRAID_CONFIGURATION_CLASSNAME, 0, "tenantId",
+            "test_tenantId");
+
+        // Configure the OIDC user class for the second user tobe able to test the switch to XWiki user functionality.
+        Object userObject =
+            setup.rest().object(new LocalDocumentReference("XWiki", SECOND_USER_NAME), "XWiki.OIDC.UserClass");
+        userObject.withProperties(TestUtils.RestTestUtils.property("issuer", "some_issuer"));
+        setup.rest().add(userObject);
+        setup.forceGuestUser();
+    }
+
+    @Order(1)
+    @Test
+    void authenticate(TestUtils testUtils)
+    {
+        // Switch the authentication service to OIDC.
+        EntraIDViewPage entraIDViewPage = new EntraIDViewPage();
+        testUtils.loginAsAdmin();
+        entraIDViewPage.goToHomePage();
+        AuthServiceViewPage authServiceViewPage = new AuthServiceViewPage();
+        authServiceViewPage.navigateToAuthenticationAdmin();
+        authServiceViewPage.switchToOIDCAuthenticationService();
+        assertTrue(authServiceViewPage.isOIDCSelected());
+        testUtils.forceGuestUser();
+
+        // Check that the login functionality correctly redirect the guest to Microsoft login page, respectively that
+        // the OIDC bypass login is working.
+        entraIDViewPage.goToHomePage();
+        entraIDViewPage.clickLogin();
+        testUtils.getDriver().waitUntilElementIsVisible(By.id("exceptionMessageContainer"));
+        WebElement loginPageMessage = entraIDViewPage.getMicrosoftContainer();
```

Comment:
You need to wait for the element inside the PO method. `getMicrosoftContainer` is wrong if you can't rely on it to return what you need.